src/core/react/ColorPicker.tsx

```diff
@@ -56,6 +57,20 @@ export const ColorPicker = React.memo((props: ColorPickerProps) => {
     const [colorHex, setColorHex] = React.useState(color.toString());
     const [visible, setVisible] = React.useState<boolean>(false);
 
+    const colorsCurrentlyUsed = useSelector((state) => Object.values(state['editor']['presentState']['state']['values']['diagrams']['items']));
+
+const colorsCurrentlyUsedInEditor = (theArray: any) => {
```

Comment:
You should just inject the colors as props in my opinion.

Reply:
Updated